File: kotlin/digitalwallet/src/main/kotlin/digitalwallet/data/common/TransactionMetadata.kt
package digitalwallet.data.common

data class BankAccountIdentification(
    val bankCode: String,
    val branchCode: String,
    val accountNumber: String,
)

data class EntityInfo(
    val name: String?,
    val bankAccount: BankAccountIdentification
)

data class TransactionMetadata(
    val counterparty: EntityInfo?,
)

File: scala/digitalwallet/app/src/main/scala/core/domain/model/TransactionMetadata.scala
package core.domain.model

case class BankAccountIdentification(
  bankCode: String,
  branchCode: String,
  accountNumber: String,
)

case class EntityInfo(
  name: Option[String],
  bankAccount: BankAccountIdentification,
)

case class TransactionMetadata(
  counterparty: Option[EntityInfo]
)
